pkg/controller/common/settings/canonical_config_test.go

```diff
@@ -451,3 +451,70 @@ func TestNewCanonicalConfigFrom(t *testing.T) {
 		})
 	}
 }
+
+func TestCanonicalConfig_HasChildConfig(t *testing.T) {
+	tests := []struct {
+		name string
+		c    *CanonicalConfig
+		key  string
+		want bool
+	}{
+		{
+			name: "nil config",
+			c:    nil,
+			key:  "x",
+			want: false,
+		},
+		{
+			name: "empty config",
+			c:    MustCanonicalConfig(map[string]interface{}{}),
+			key:  "x",
+			want: false,
+		},
+		{
+			name: "valid top-level key but not a child config",
+			c: MustCanonicalConfig(map[string]interface{}{
+				"x": "y",
+			}),
+			key:  "x",
+			want: false,
+		},
+		{
+			name: "valid top level key",
+			c: MustCanonicalConfig(map[string]interface{}{
+				"x": map[string]interface{}{
+					"y": "1",
+					"z": "2",
+				},
+			}),
+			key:  "x",
+			want: true,
+		},
+		{
+			name: "valid nested  key",
+			c: MustCanonicalConfig(map[string]interface{}{
+				"x": map[string]interface{}{
+					"y": map[string]interface{}{},
+					"z": "2",
+				},
+			}),
+			key:  "x.y",
+			want: true,
+		},
+		{
+			name: "absent key",
+			c: MustCanonicalConfig(map[string]interface{}{
+				"x": "y",
+			}),
+			key:  "z",
+			want: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := tt.c.HasChildConfig(tt.key); got != tt.want {
+				t.Errorf("HasChildConfig() = %v, want %v", got, tt.want)
+			}
+		})
+	}
```

Comment:
If you accept the previous suggestion about checking the error:

```suggestion
		{
			name: "invalid config",
			c: MustCanonicalConfig(map[string]interface{}{
				"x": "{",
			}),
			key:     "x",
			want:    false,
			wantErr: true,
		},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			got, err := tt.c.HasChildConfig(tt.key)
			if (err != nil) != tt.wantErr {
				t.Errorf("HasChildConfig() error = %v, wantErr %v", err, tt.wantErr)
				return
			}
			if got != tt.want {
				t.Errorf("HasChildConfig() = %v, want %v", got, tt.want)
			}
		})
	}
```

Reply:
I am still facing this issue on a standalone docker environment , when I specify this inenterprise-search.yml  (version 8.1.2)
``````
ent_search.auth:
  native1:
    source: elasticsearch-native
    order: 0
``````

Here is the error message 

workspace-kibana-1         | [2022-04-04T03:22:42.080+00:00][FATAL][root] Error: Unknown configuration key(s): "ent_search.auth.native1.source", "ent_search.auth.native1.order". Check for spelling errors and ensure that expected plugins are installed.
workspace-kibana-1         |     at ensureValidConfiguration (/usr/share/kibana/src/core/server/config/ensure_valid_configuration.js:35:11)
workspace-kibana-1         |     at Server.preboot (/usr/share/kibana/src/core/server/server.js:164:5)
workspace-kibana-1         |     at Root.preboot (/usr/share/kibana/src/core/server/root/index.js:48:14)
workspace-kibana-1         |     at bootstrap (/usr/share/kibana/src/core/server/bootstrap.js:99:9)
workspace-kibana-1         |     at Command.<anonymous> (/usr/share/kibana/src/cli/serve/serve.js:216:5)